modules/packages/LinearAlgebra.chpl

```diff
@@ -1313,6 +1318,28 @@ proc lu (A: [?Adom] ?eltType) {
   return (LU,ipiv);
 }
 
+
+proc lu (A: [?Adom] ?t) where (usingLAPACK && isLAPACKType(t)){
+
+  if Adom.rank != 2 then
+    halt("Wrong rank for LU factorization");
+
+  if isDistributed(A) then
+    compilerError("LU factorization does not support distributed vectors/matrices");
+  
+  use SysCTypes;
+  use LAPACK;
+  var A_clone = A;
+
+  var rows = Adom.shape[0];
+  var cols = Adom.shape[1];
+  var ipiv: [{1..rows}] c_int;
+  var info = getrf(lapack_memory_order.row_major,A_clone,ipiv);
+  var U = triu(A_clone);
+  var L = A_clone-U+eye(A.domain);
+  return(L[1..rows,1..rows],U,ipiv);
```

Comment:
This return type differs from the other overload, right? I don't think that will work as is, so we'll either need to determine that we are changing the return type in https://github.com/chapel-lang/chapel/issues/16973 or combine `L` and `U` here.